pymatgen/command_line/bader_caller.py

```diff
@@ -193,3 +206,175 @@ def get_oxidation_state_decorated_structure(self):
                    for i in range(len(structure))]
         structure.add_oxidation_state_by_site(charges)
         return structure
+
+    @property
+    def result(self):
+
+        summary = {
+            "min_dist": [d['min_dist'] for d in self.data],
+            "charge": [d['charge'] for d in self.data],
+            "volume": [d['atomic_vol'] for d in self.data],
+            "vacuum_charge": self.vacuum_charge,
+            "vacuum_volume": self.vacuum_volume,
+            "reference_used": self.reference_used,
+            "bader_version": self.version,
+        }
+
+        if self.potcar:
+            charge_transfer = [self.get_charge_transfer(i) for i in range(len(self.data))]
+            summary['charge_transfer'] = charge_transfer
+
+        return BaderResult(summary)
+
+
+class BaderResult(MSONable):
```

Comment:
Ideally, I would have made the BaderAnalysis itself MSONable, but I can't without changing the constructor and breaking backwards compatibility. I put it into its own class under the assumption it'd have some methods added in future (e.g. like `get_oxidation_state_decorated_structure`), but I'm more than happy to just put `from_path` and `from_object` into their own functions for now instead.

Will modify `from_path` to take a `suffix` argument and glob those, and if `suffix==None` will do a sort as proposed.